terraform_compliance/extensions/terraform.py

```diff
@@ -345,7 +345,7 @@ def _mount_references(self):
                 if ref_list:
                     ref_type = self.configuration['resources'][resource]['expressions'].get('type', {})
 
-                    if not ref_type and not self.is_type(resource, 'data'):
+                    if (not ref_type or not isinstance(ref_type, str)) and not self.is_type(resource, 'data'):
                         resource_type, resource_id = resource.split('.')
                         ref_type = resource_type
 
```

Comment:
`ref_type = self.configuration['resources'][resource]['expressions'].get('type', {})` may occasionally return a bad type (e.g. a non empty dictionary).

In those cases, play safe and use the general case instead of further parsing.

Reply:
* I'm not completely clear on what `ref_type` should always be. Why are we not pulling it directly from `ref_list`?
* From my limited observation, `ref_type` at line 346 seems to return `{}` more often than not. Would it be better to set `ref_type` to `resource_type` right away, and try to cover edge cases (`ref_type` is not a resource) below that?